lepton/image.go

```diff
@@ -225,19 +241,43 @@ func buildImage(c *Config, m *Manifest) error {
 		return errors.Wrap(err, 1)
 	}
 
+	bootImageOpen, err := os.Open(c.Boot)
+	if err != nil {
+		return errors.Wrap(err, 1)
+	}
+	defer bootImageOpen.Close()
+
 	// produce final image, boot + kernel + elf
+	if c.DiskImage == "" {
+		finalImagePath, err := GetImagePathByProgramPath(programAbsPath)
+		if err != nil {
+			return errors.Wrap(err, 1)
+		}
+		c.DiskImage = filepath.Join(finalImagePath, "image")
+	}
 	fd, err := createFile(c.DiskImage)
-	defer fd.Close()
 	if err != nil {
 		return errors.Wrap(err, 1)
 	}
-	catcmd := exec.Command("cat", c.Boot, mergedImg)
-	catcmd.Stdout = fd
-	err = catcmd.Start()
-	if err != nil {
+	defer fd.Close()
+
+	if _, err = io.Copy(fd, bootImageOpen); err != nil {
+		return errors.Wrap(err, 1)
+	}
+	if _, err = io.Copy(fd, mergedImg); err != nil {
 		return errors.Wrap(err, 1)
 	}
-	catcmd.Wait()
+
+	if err := mergedImg.Close(); err != nil {
+		return errors.Wrap(err, 1)
+	}
+
+	if len(c.Debugflags) == 0 {
+		if err := RemoveTMPPathByProgramPath(programAbsPath); err != nil {
```

Comment:
Not sure i understand this, how is removing temp related to debug flags?

Reply:
Ah, also forgot to mention that I added the following:
* If we have debug option specified than tempimage will remain
* If nothing debug specified that tempimage is removed along with it's temp directory at `$HOME/.ops/<temp_path>`

Reply:
As @eyberg said that `tempimage` is more for debug purposes...

Reply:
Sorry for the confusion, tempimage is an artifact of how we build the images. First we merge kernel an user executable to a file (which is the tempimage), then we merge tempimage and the boot loader to final image. So you can remove the temp-image right after the final merge.

Reply:
I see.. but again, for debugging purposes maybe we can leave it if user specifies debug option?

Reply:
i don't see a need for tempimage or any of the other artifacts to be present -- '-d' works fine w/the resulting built image - for instance in the old model -- 'image' we want to keep but anything in ~/.staging we don't want to keep

Reply:
Actually tempimage is not stored in .staging, everything is under ~/.ops, but Only in case someone specified -d option, tempimage is saved, in other cases it’s removed

Reply:
tempimage does not need to be saved at all -- '-d' only uses the built image - we can completely remove tempimage

Reply:
Ah, okay, I’ll update PR to always remove it

Reply:
Updated PR to always remove tempimage